src/main/core.js

```diff
@@ -58,7 +58,7 @@ function attachComments(text, ast, opts) {
     comments.attach(astComments, ast, text, opts);
   }
   ast.tokens = [];
-  opts.originalText = text.trimRight();
+  opts.originalText = opts.parser === "yaml" ? text : text.trimRight();
```

Comment:
I'm not sure why we need to `trimRight` here but removing it will cause other languages to fail so I just disable it for yaml as a temporary workaround, we should fix it in the future.